### PR TITLE
Give message with line and column of CSV file if data conversion fails. Allow parsing of DateTime data, and do the basic arithmetic on them

### PR DESCRIPTION
**Pull request by derekdiamond**
We are excited to review your PR.

So we can do the best job, please check:

- [x] There's a descriptive title that will make sense to other developers some time from now. 
- [x] There's associated issues. All PR's should have issue(s) associated - unless a trivial self-evident change such as fixing a typo. You can use the format `Fixes #nnnn` in your description to cause GitHub to automatically close the issue(s) when your PR is merged.
- [x] Your change description explains what the change does, why you chose your approach, and anything else that reviewers should know.
- [x] You have included any necessary tests in the same PR.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/machinelearning?pullRequest=5791) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/machinelearning?pullRequest=5791) before we can accept your contribution.<br/><br/>:x: derekdiamond [sign now](https://cla.dotnetfoundation.org/dotnet/machinelearning?pullRequest=5791)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/machinelearning?pullRequest=5791) it.</sub>

**Comment by michaelgsharp** [comment]
Hey @derekdiamond, thanks for taking the time to submit this PR!

It looks like somehow some of the compiled native files were placed in src/native/out. Probably a build mess up, but can you remove those files from the PR?

**Comment by derekdiamond** [comment]
> Hey @derekdiamond, thanks for taking the time to submit this PR!
> 
> It looks like somehow some of the compiled native files were placed in src/native/out. Probably a build mess up, but can you remove those files from the PR?

Sure will do

**Comment by michaelgsharp** [comment]
@pgovind since this is in `DataAnalysis` can you take a look? I'm still not very familiar with the new code.

**Comment by pgovind** [comment]
Yup, this just popped up in my notifications. Will review in a bit

File: src/Microsoft.Data.Analysis/DateTimeComputation.cs [hunk 1]
```diff
@@ -0,0 +1,310 @@
+﻿using System;
```
Comment (pgovind) [inline src/Microsoft.Data.Analysis/DateTimeComputation.cs:1]: This will need the license header. Copying from another file should work just fine

**Review by pgovind — COMMENTED** [review]
Left a comment. This is great work, this PR is almost ready to go in. I would just like to see 2 things:

1. Some unit tests. Since we're now adding `DateTime` to be detected automatically by LoadCsv, we should test that path. I think duplicating one of the existing LoadCsv tests and adding a `DateTime` column to it will work fine. 
2. There are still some CMAKE build artifacts in the changed files. I think you can just get rid of them.

**Comment by derekdiamond** [comment]
Cool will do

**Comment by pgovind** [comment]
> Cool will do

Yup I see that the CMAKE artifacts are gone now. Not sure if you intended to fix the other comments with the last commit. Just calling it out in case you has commits locally that didn't get pushed

**Comment by derekdiamond** [comment]
Thanks, all done added tests as well for IO and Computations. So hope all is good now!

**Comment by pgovind** [comment]
> So hope all is good now!

Almost :) 

1. Looks like the cmake stuff creeped in again. (not related to this PR: it looks like all the CMAKE stuff is build artifacts. Maybe consider adding out/* to your .gitignore? But remember not to push the .gitignore file itself!)
2. Copy paste the license text into DateTimeComputation.cs from any other file that has it?

**Comment by derekdiamond** [comment]
I'm getting there!. Hope all the out\* artifacts are gone!

**Comment by pgovind** [comment]
Since we're closing to having a new preview release of ML, I cleaned up the remaining 4 build artifacts in https://github.com/dotnet/machinelearning/pull/5834. Going to close this PR in favor of that one.

Awesome work here! Thanks @derekdiamond

**Pull request closed without merging**